common/numerated/src/interval.rs

```diff
@@ -85,352 +67,251 @@ impl<T: Numerated> NonEmptyInterval<T> {
         self.end
     }
 
-    /// Converts to [Interval], which implements iterator.
-    pub fn iter(&self) -> Interval<T> {
+    /// Converts to [IntervalIterator].
+    pub fn iter(&self) -> IntervalIterator<T> {
         (*self).into()
     }
 
     /// Into (start, end)
-    pub fn into_inner(self) -> (T, T) {
+    pub fn into_parts(self) -> (T, T) {
         self.into()
     }
-}
 
-/// Error which occurs when trying to convert empty [Interval] into [NonEmptyInterval].
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct IntoNonEmptyIntervalError;
-
-impl<T: Numerated> TryFrom<&Interval<T>> for NonEmptyInterval<T> {
-    type Error = IntoNonEmptyIntervalError;
-
-    fn try_from(interval: &Interval<T>) -> Result<Self, Self::Error> {
-        interval
-            .end
-            .map(|end| unsafe {
-                // Guaranteed by `Self` that start ≤ end
-                NonEmptyInterval::new_unchecked(interval.start, end)
+    /// Returns new [Interval] with `start` = `start` + 1, if it's possible.
+    pub fn inc_start(&self) -> Option<Self> {
```

Comment:
`inc_start(&mut self) -> Result<()>`
or
`with_increased_start(self) -> Option<Self>`

Reply:
> No need for result for such simple action. `with_` usually used for constructor or builder pattern

why don't you build it here

Reply:
Because type itself is not builder

Reply:
agree with @ark0f